src/main.c

```diff
@@ -1141,8 +1214,19 @@ static int on_config_listen(h2o_configurator_command_t *cmd, h2o_configurator_co
                             return -1;
                         }
                     } else {
-                        if ((fd = open_tcp_listener(cmd, node, hostname, servname, ai->ai_family, ai->ai_socktype, ai->ai_protocol,
-                                                    ai->ai_addr, ai->ai_addrlen)) == -1) {
+#if defined(__linux__) && defined(SO_REUSEPORT)
+                        /**
+                         * socket open should after setuid() call when so_reuseport enabled in case of "user" has been configured
+                         * because: To prevent port hijacking all sockets bound to the same port using so_reuseport must have the
+                         * same uid
```

Comment:
I think that we need to open file descriptors that listens for incoming connections _before_ calling `setuid`. Otherwise, the server cannot bind to privileged ports.

One way to do that would be to create sockets _for every thread_ during configuration, store them in an array of file descriptors in `listener_config_t`, and then distribute those socket at the beginning of `run_loop`.